hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/NativeSparkDataSet.java

```diff
@@ -821,6 +821,58 @@ private String planIfLast(AbstractSpliceFunction f, boolean isLast) {
         return (plan != null && !plan.isEmpty() ? plan : f.getSparkName());
     }
 
+    public static boolean joinProjectsOnlyLeftTableColumns(JoinType joinType) {
+        return (joinType == DataSet.JoinType.LEFTANTI ||
+                joinType == DataSet.JoinType.LEFTSEMI);
+    }
+    private Dataset<Row> fixupColumnNames(JoinOperation op, JoinType joinType,
+                                          Dataset<Row> leftDF, Dataset<Row> rightDF,
+                                          Dataset<Row> joinedDF) {
+        if (!joinProjectsOnlyLeftTableColumns(joinType) &&
```

Comment:
Can you create a new predicate method that would describe what the complex `if` uses to execute the `true` body? That would make comprehension slightly easier.

Since the entire body is pretty much the `true` block of the `if`, I'd recommend negate the `if` condition and `return joinedDF`. With such `if` you could execute the `if` body outside `if`.

Reply:
@jaceklaskowski Thanks for your comments.  I will incorporate the changes in my next push.